EIPS/eip-1363.md

```diff
@@ -181,24 +181,27 @@ interface ERC1363Spender {
    * approval. Return of other than the magic value MUST result in the
    * transaction being reverted.
    * Note: the token contract address is always the message sender.
-   * @param owner address The address which called `approveAndCall` function
-   * @param value uint256 The amount of tokens to be spent
+   * @param sender address The address which called `approveAndCall` function
+   * @param amount uint256 The amount of tokens to be spent
    * @param data bytes Additional data with no specified format
    * @return `bytes4(keccak256("onApprovalReceived(address,uint256,bytes)"))`
    *  unless throwing
    */
-  function onApprovalReceived(address owner, uint256 value, bytes memory data) external returns (bytes4);
+  function onApprovalReceived(address sender, uint256 amount, bytes memory data) external returns (bytes4);
 }
 ``` 
 
+Note that `transferAndCall`, `transferFromAndCall` MUST revert if the recipient is an EOA address, because EOA recipients do not implement the required ERC1363Receiver interface.  
+Note that `approveAndCall` MUST revert if the spender is an EOA address, because EOA spenders do not implement the required ERC1363Spender interface.
+
```

Comment:
I think it is important to be explicit about the following point here, even though the correct order is obvious in the NatSpec comments:

> `transferAndCall` and `transferFromAndCall` MUST call the `onTransferReceived` method in the receiver only AFTER updating the token contract's internal state, such as transferring the requested amount from the sender's balance to the recipient's balance, and reducing the ERC20 allowance by the approved amount (in the case of `transferFromAndCall`), in order to eliminate the chance of reentrancy attacks.

> `approveAndCall` MUST call the `onApprovalReceived` method in the spender only AFTER all updating the token contract's internal state, such as granting the spender the approved allowance for spending the sender's tokens, in order to eliminate the chance of reentrancy attacks.

Reply:
Maybe even also be explicit about the fact that the transaction should revert if a call to `onTransferReceived` or `onApprovalReceived` reverts. This is important because there is a big move towards using `address.call` now for calling external contracts, and `call` does not automatically revert if the call fails for some reason.

Reply:
These are already expressed in code comments 

```
   * @dev Any ERC1363 smart contract calls this function on the recipient
   * after a `transfer` or a `transferFrom`. This function MAY throw to revert and reject the
   * transfer. Return of other than the magic value MUST result in the
   * transaction being reverted.`
   ```

```
   * @dev Any ERC1363 smart contract calls this function on the recipient
   * after an `approve`. This function MAY throw to revert and reject the
   * approval. Return of other than the magic value MUST result in the
   * transaction being reverted.
   ```

Don't want to make the spec so repetitive.